yt_dlp/extractor/iqiyi.py

```diff
@@ -392,3 +405,320 @@ def _real_extract(self, url):
+            r'\(({[^}]*})\[\w+\][^\)]*\)\s*\+\s*["\']\.["\']\s*\+\s*({[^}]*})\[\w+\]\+["\']\.js', webpack_js, 'JS locations', group=(1, 2))]
+        for module_index in reversed(webpack_map2.keys()):
+            module_js = self._download_webpage(
+                f'https://stc.iqiyipic.com/_next/static/chunks/{webpack_map1.get(module_index, module_index)}.{webpack_map2[module_index]}.js',
+                video_id, note=f'Downloading #{module_index} module JS', errnote='Unable to download module JS', fatal=False) or ''
+            if 'vms request' in module_js:
+                self._downloader.cache.store('iq', 'player_js', module_js)
+                return module_js
+        raise ExtractorError('Unable to extract player JS')
+
+    def _extract_cmd5x_function(self, webpage, video_id):
+        return self._search_regex(r',\s*(function\s*\([^\)]*\)\s*{\s*var _qda.+_qdc\(\)\s*})\s*,',
+                                  self._extract_vms_player_js(webpage, video_id), 'signature function')
+
+    def _update_bid_tags(self, webpage, video_id):
+        extracted_bid_tags = self._parse_json(
+            self._search_regex(
+                r'arguments\[1\][^,]*,\s*function\s*\([^\)]*\)\s*{\s*"use strict";?\s*var \w=({.+}})\s*,\s*\w\s*=\s*{\s*getNewVd',
+                self._extract_vms_player_js(webpage, video_id), 'video tags', default=''),
+            video_id, transform_source=js_to_json, fatal=False)
+        if not extracted_bid_tags:
+            return
+        self._BID_TAGS = {
+            bid: traverse_obj(extracted_bid_tags, (bid, 'value'), expected_type=str_or_none, default=self._BID_TAGS.get(bid))
+            for bid in extracted_bid_tags.keys()
+        }
+
+    def _real_extract(self, url):
+        video_id = self._match_id(url)
+        webpage = self._download_webpage(url, video_id)
+        self._update_bid_tags(webpage, video_id)
+
+        next_props = self._search_nextjs_data(webpage, video_id)['props']
+        page_data = next_props['initialState']['play']
+        video_info = page_data['curVideoInfo']
+        tvid, vid = video_info['tvId'], video_info['vid']
+        src = traverse_obj(next_props, ('initialProps', 'pageProps', 'ptid'), expected_type=str_or_none, default='01010031010018000000')
+        get_cookie = lambda name, default: self._get_cookies('https://iq.com/')[name].value if self._get_cookies('https://iq.com').get(name) else default
+        dfp, mode, lang = get_cookie('dfp', ''), get_cookie('mod', 'intl'), get_cookie('lang', 'en_us')
+        js_bid_list = '[' + ','.join(['0', *self._BID_TAGS.keys()]) + ']'
+        cmd5x_func = self._extract_cmd5x_function(webpage, video_id)
+
+        # bid 0 as an initial format checker
+        dash_paths = self._parse_json(PhantomJSwrapper(self).get(url, html='<!DOCTYPE html>', video_id=video_id, note2='Executing signature code', jscode="""
```

Comment:
I suppose the native JSInterpreter as enhanced recently doesn't cut it?

Reply:
AFAIK JSInterpreter doesn't include DOM emulation which the function explicitly checks